clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/view/provider/KubernetesDataProviderIntegrationTest.java

```diff
@@ -0,0 +1,1117 @@
+import com.netflix.spinnaker.clouddriver.kubernetes.description.manifest.KubernetesKind;
+import com.netflix.spinnaker.clouddriver.kubernetes.names.KubernetesManifestNamer;
+import com.netflix.spinnaker.clouddriver.kubernetes.op.handler.KubernetesDeploymentHandler;
+import com.netflix.spinnaker.clouddriver.kubernetes.op.handler.KubernetesHandler;
+import com.netflix.spinnaker.clouddriver.kubernetes.op.handler.KubernetesPodHandler;
+import com.netflix.spinnaker.clouddriver.kubernetes.op.handler.KubernetesReplicaSetHandler;
+import com.netflix.spinnaker.clouddriver.kubernetes.op.handler.KubernetesServiceHandler;
+import com.netflix.spinnaker.clouddriver.kubernetes.op.handler.KubernetesUnregisteredCustomResourceHandler;
+import com.netflix.spinnaker.clouddriver.kubernetes.op.handler.ManifestFetcher;
+import com.netflix.spinnaker.clouddriver.kubernetes.op.job.KubectlJobExecutor;
+import com.netflix.spinnaker.clouddriver.kubernetes.security.GlobalKubernetesKindRegistry;
+import com.netflix.spinnaker.clouddriver.kubernetes.security.KubernetesKindRegistry;
+import com.netflix.spinnaker.clouddriver.kubernetes.security.KubernetesNamedAccountCredentials;
+import com.netflix.spinnaker.clouddriver.kubernetes.security.KubernetesSelectorList;
+import com.netflix.spinnaker.clouddriver.kubernetes.security.KubernetesV2Credentials;
+import com.netflix.spinnaker.clouddriver.model.Application;
+import com.netflix.spinnaker.clouddriver.model.HealthState;
+import com.netflix.spinnaker.clouddriver.model.Instance;
+import com.netflix.spinnaker.clouddriver.model.LoadBalancerInstance;
+import com.netflix.spinnaker.clouddriver.model.LoadBalancerProvider.Details;
+import com.netflix.spinnaker.clouddriver.model.LoadBalancerServerGroup;
+import com.netflix.spinnaker.clouddriver.model.ServerGroup;
+import com.netflix.spinnaker.clouddriver.model.ServerGroupManager.ServerGroupManagerSummary;
+import com.netflix.spinnaker.clouddriver.model.ServerGroupSummary;
+import com.netflix.spinnaker.clouddriver.names.NamerRegistry;
+import com.netflix.spinnaker.clouddriver.search.SearchResultSet;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsRepository;
+import com.netflix.spinnaker.clouddriver.security.MapBackedAccountCredentialsRepository;
+import com.netflix.spinnaker.kork.configserver.CloudConfigResourceService;
+import com.netflix.spinnaker.kork.configserver.ConfigFileService;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+import org.mockito.internal.stubbing.defaultanswers.ReturnsSmartNulls;
+
+@ExtendWith(SoftAssertionsExtension.class)
```

Comment:
You no doubt pondered this, but if the goal is to wire things up the way spring does, why not use e.g. `@ExtendWith(SpringExtension.class)` like https://github.com/spinnaker/echo/blob/ac270b219511fb16dd142ec32d2aa8b8a961cd6b/echo-web/src/test/java/com/netflix/spinnaker/echo/telemetry/TelemetrySpec.java does?  Maybe some `@Autowired` simplifies getting the objects required for testing?

Reply:
Thanks for the pointer!  In this case I want a bit more control over how each of these beans are built than is possible (or that I've found easy to do with spring integration tests).  For example, to keep things simple I only wanted it to know about 4 of the `*Handler` classes rather than everything on the classpath.

In some sense, this is aspirational as well...in that I don't think I *should* need ~50 lines of code to get a functioning `KubernetesV2Credentials`, so hopefully this will get simpler with future refactors (whereas moving to use Spring would be somewhat admitting defeat).

But I do agree given how complex it is now, it's 50/50 whether wiring this up manually vs. using spring would have been easier.

Reply:
Oooph.  I just looked at [KubernetesV2Credentials.Factory](https://github.com/spinnaker/clouddriver/blob/36235837d452f251a4f50c6bf3d876e6eabdec9c/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesV2Credentials.java#L690).  No fun.  Yeah, I agree doing it the way you did will it make it more clear if things get simpler...and gives you more control.